### ISSUE TITLE
rake db:migrate broken

### ISSUE THREAD

**Issue by beanieboi** [rails-girls-summer-of-code/rgsoc-teams#375]
hey

i just tried to setup the teams app and the migrations are not running through.

```
undefined method `project_proposals_open_at' for #<Season:0x007fe1784709f8>/Users/ben/Development/rgsoc-teams/app/models/season.rb:96:in `set_application_dates'
/Users/ben/Development/rgsoc-teams/app/models/season.rb:9:in `current'
/Users/ben/Development/rgsoc-teams/db/migrate/20150625085615_add_seasons_to_mailings.rb:4:in `change'
NoMethodError: undefined method `project_proposals_open_at' for #<Season:0x007fe1784709f8>
/Users/ben/Development/rgsoc-teams/app/models/season.rb:96:in `set_application_dates'
/Users/ben/Development/rgsoc-teams/app/models/season.rb:9:in `current'
/Users/ben/Development/rgsoc-teams/db/migrate/20150625085615_add_seasons_to_mailings.rb:4:in `change'
Tasks: TOP => db:migrate
(See full trace by running task with --trace)
```

the problem is that in `20150625085615_add_seasons_to_mailings.rb` we try to access `Season.current` which tried to create a Season with attributes `project_proposals_open_at` that are not present yet in the current point of migrations.

what do you think would be a good fix? i can work on it!
what i usually do is to squash the migrations. there is no point in having 120 migration files. i would usually squash all migrations that are older than 1-2 months in 1 big migrations. this allows you to still revert deployments and do down migrations. i doubt that there is value in really old migration files.

let me know what you think about this. happy to work on any solution.

best
ben

**Comment by beanieboi** [comment]
i can also move the line `Mailing.update_all(seasons: [Season.current.name])` from 
`20150625085615_add_seasons_to_mailings.rb` to `20151205133354_add_project_proposal_dates_to_seasons.rb`

it alters the migrations, but i think there is no way to fix this without changing the migrations.

**Comment by F3PiX** [comment]
Just pulled upstream and run db:migrate. I did not have this migration problem. Does that prove anything at all? :-)

**Comment by beanieboi** [comment]
hey @F3PiX 

did you start with a empty database? when i changed the two files like mentioned above it works for me. it's important that the db is empty and you run all migrations.

**Comment by F3PiX** [comment]
Aha, no, it probably had to do only the latest migration. Thanks for clarifying.

**Comment by carpodaster** [comment]
@beanieboi That is indeed a bug! @ramonh ran into this the other day. We "fixed" it by importing a recent dev DB dump but that's obviously a real solution for the problem.

Squashing old migrations sounds like a very good idea! :+1: Thanks for working on this, it's highly appreciated! :confetti_ball:

**Comment by carpodaster** [comment]
@beanieboi Just an idea in the middle of something else: can't you get past this error by doing `rake db:create; rake db:schema:load` ?

**Comment by beanieboi** [comment]
loading the schema works. It's just updating the mailings where we call Season.current that breaks everything.

**Comment by carpodaster** [comment]
But if you loaded the schema on a fresh DB, you don't have to run the migrations. And there aren't any mailings to update anyway.

**Comment by beanieboi** [comment]
That's true. But we should fix that bug since beginners can run into it and don't know what do to. db:migrate is part of every rails tutorial and it should work IMHO

As I wrote in the issue we can also just switch the mailing update to a later migration where the Season schema is complete.

**Comment by carpodaster** [comment]
(Hopefully) fixed in 1ec0cacd6d26f4e0fd07a68be03d69251c41b7ec
